core/templates/components/state-editor/state-editor.directive.ts

```diff
@@ -130,6 +140,16 @@ angular.module('oppia').directive('stateEditor', [
             $scope.interactionIdIsSet = false;
             $scope.servicesInitialized = false;
             $scope.stateName = StateEditorService.getActiveStateName();
+            ctrl.explorationId = ContextService.getExplorationId();
+            ctrl.explorationUrl = UrlInterpolationService.interpolateUrl(
+              '/createhandler/data/<exploration_id>', {
+                exploration_id: ctrl.explorationId
+              });
+            $http.get(ctrl.explorationUrl)
+              .then(function(response) {
+                $scope.explorationIsLinkedToStory = (
+                  response.data.exploration_is_linked_to_story);
+              });
```

Comment:
Why is this call being made in the state-editor? It would fetch the entire exp data every time the state editor is initialised.

Can we set this boolean when the exploration data is fetched the first time and use that here?

-

Reply:
Done. PTAL!